cmd/juju/commands/bootstrap.go

```diff
@@ -315,31 +315,290 @@ func (c *bootstrapCommand) Run(ctx *cmd.Context) (resultErr error) {
 	if err := c.parseConstraints(ctx); err != nil {
 		return err
 	}
+
+	// Start by checking for usage errors, requests for information
+	finished, err := c.handleCommandLineErrorsAndInfoRequests(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if finished {
+		return nil
+	}
+
+	// Run interactive bootstrap if needed/asked for
+	if c.interactive {
+		if err := c.runInteractive(ctx); err != nil {
+			return errors.Trace(err)
+		}
+		// now run normal bootstrap using info gained above.
+	}
+
+	cloud, err := c.cloud(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	provider, err := environs.Provider(cloud.Type)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// Custom clouds may not have explicitly declared support for any auth-
+	// types, in which case we'll assume that they support everything that
+	// the provider supports.
+	if len(cloud.AuthTypes) == 0 {
+		for authType := range provider.CredentialSchemas() {
+			cloud.AuthTypes = append(cloud.AuthTypes, authType)
+		}
+	}
+
+	credentials, regionName, err := c.credentialsAndRegionName(ctx, cloud)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	region, err := getRegion(cloud, c.Cloud, regionName)
+	if err != nil {
+		fmt.Fprintf(ctx.GetStderr(),
+			"%s\n\nSpecify an alternative region, or try %q.\n",
```

Comment:
This is a user prompt though so the output that's needed is different to what annotate would print.